src/Microsoft.Restier.EntityFramework/DbApi.cs

```diff
@@ -56,71 +57,40 @@ protected T DbContext
         /// </returns>
         protected override ApiBuilder ConfigureApi(ApiBuilder builder)
         {
-            builder = base.ConfigureApi(builder);
-            builder.CutoffPrevious<IModelBuilder>(ModelProducer.Instance);
-            builder.CutoffPrevious<IModelMapper>(new ModelMapper(typeof(T)));
-            builder.CutoffPrevious<IQueryExpressionSourcer, QueryExpressionSourcer>();
-            builder.CutoffPrevious<IQueryExecutor>(QueryExecutor.Instance);
-            builder.CutoffPrevious<IChangeSetPreparer, ChangeSetPreparer>();
-            builder.CutoffPrevious<ISubmitExecutor>(SubmitExecutor.Instance);
-            return builder;
-        }
-
-        /// <summary>
-        /// Creates the API context for this API.
-        /// </summary>
-        /// <param name="configuration">
-        /// The API configuration to use.
-        /// </param>
-        /// <returns>
-        /// The API context for this API.
-        /// </returns>
-        protected override ApiContext CreateApiContext(
-            ApiConfiguration configuration)
-        {
-            var context = base.CreateApiContext(configuration);
-            var dbContext = this.CreateDbContext();
+            builder = base.ConfigureApi(builder)
+                .CutoffPrevious<IModelBuilder>(ModelProducer.Instance)
+                .CutoffPrevious<IModelMapper>(new ModelMapper(typeof(T)))
+                .CutoffPrevious<IQueryExpressionSourcer, QueryExpressionSourcer>()
+                .CutoffPrevious<IQueryExecutor>(QueryExecutor.Instance)
+                .CutoffPrevious<IChangeSetPreparer, ChangeSetPreparer>()
+                .CutoffPrevious<ISubmitExecutor>(SubmitExecutor.Instance);
+            builder.Services
+                .AddScoped<T>(sp =>
```

Comment:
This is also to ease those service that already know in which Api type they're used. If you think unnecessary I'll remove it.
And actually I want another design discussion here.
Should we just register T normally to use its constructor to instantiate? And of course CreateDbContext should be removed.
- Subclass could still override by registering T in ConfigApi.
- The problem is `ProxyCreationEnabled = false`. Maybe we can leave that to developer?